openvdb_points/tools/PointConversion.h

```diff
@@ -212,6 +213,44 @@ class PointAttributeVector {
 
 namespace point_conversion_internal {
 
+
+// ConversionTraits to create the relevant Attribute Handles from a LeafNode
+template <typename T> struct ConversionTraits
+{
+    typedef AttributeHandle<T> Handle;
+    typedef AttributeWriteHandle<T> WriteHandle;
+    static T zero() { return T(0); }
+    template <typename LeafT>
+    static typename Handle::Ptr handleFromLeaf(LeafT& leaf, Index index) {
+        const AttributeArray& array = leaf.template constAttributeArray(index);
+        return Handle::create(array);
+    }
+    template <typename LeafT>
+    static typename WriteHandle::Ptr writeHandleFromLeaf(LeafT& leaf, Index index) {
+        AttributeArray& array = leaf.template attributeArray(index);
+        return WriteHandle::create(array);
+    }
+}; // ConversionTraits
+template <> struct ConversionTraits<openvdb::Name>
+{
+    typedef StringAttributeHandle Handle;
+    typedef StringAttributeWriteHandle WriteHandle;
+    static openvdb::Name zero() { return ""; }
+    template <typename LeafT>
+    static typename Handle::Ptr handleFromLeaf(LeafT& leaf, Index index) {
+        const AttributeArray& array = leaf.template constAttributeArray(index);
```

Comment:
Why does this need to be templated on the leaf type? Also is the .template extension needed here?